### ISSUE TITLE
[Module Proposal]: `avm-ptn-botservice-teamsapp`

### ISSUE THREAD

**Issue by salujamanish** [Azure/Azure-Verified-Modules#1408]
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes.

### Bicep or Terraform?

Terraform

### Module Classification?

Pattern Module

### Module Name

avm-ptn-botservice-teamsapp

### Module Details

Creates an App Registration in AAD
Create AppInsights for the Bot
Azure Storage that hosts Tab static web site and Bot chat history
Create Compute resources for the Bot Web App
Web App that hosts Bot
Configure the Bot Web App settings
Register Bot web service as a bot with the Bot Framework
References-
https://github.com/microsoft/teams-copilot-starter/blob/main/teamsapp.yml
https://github.com/microsoft/teams-copilot-starter/blob/main/infra/azure.bicep

### Do you want to be the owner of this module?

Yes

**Comment by salujamanish** [comment]
last comment I saw was AVM core team is reviewing the module proposal and wait to start development until the proposal get reviewed.

Could AVM core team please update if proposal has go-ahead?

**Comment by prjelesi** [comment]
This will be pattern module and will be on hold for now.

**Comment by jaredfholgate** [comment]
@salujamanish apologies for the huge delay. Would you still like to be the owner of this module?

**Comment by jeanchg** [comment]
I just had confirmation @salujamanish won't be able to contribute and own this module anymore.

**Comment by arjenkins25** [comment]
Hi, I was interested in working on this module and being the owner if possible?

**Comment by matebarabas** [comment]
Hi @araybiajenkins,

Thanks for requesting/proposing to be an AVM module owner!

We just want to confirm **you agree to the below pages** that define what module ownership means:

- [Team Definitions & RACI](https://azure.github.io/Azure-Verified-Modules/specs/shared/team-definitions)
- [Module Specifications](https://azure.github.io/Azure-Verified-Modules/specs/module-specs)
- [Module Support](https://azure.github.io/Azure-Verified-Modules/help-support/module-support)

Any questions or clarifications needed, let us know!

If you agree, please just **reply to this issue with the exact sentence below** (as this helps with our automation 👍):

"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

Thanks,

The AVM Core Team

#RR

**Comment by araybiajenkins** [comment]
> Hi [@arjenkins25](https://github.com/arjenkins25),
> 
> Thanks for requesting/proposing to be an AVM module owner!
> 
> We just want to confirm **you agree to the below pages** that define what module ownership means:
> 
> * [Team Definitions & RACI](https://azure.github.io/Azure-Verified-Modules/specs/shared/team-definitions)
> * [Module Specifications](https://azure.github.io/Azure-Verified-Modules/specs/module-specs)
> * [Module Support](https://azure.github.io/Azure-Verified-Modules/help-support/module-support)
> 
> Any questions or clarifications needed, let us know!
> 
> If you agree, please just **reply to this issue with the exact sentence below** (as this helps with our automation 👍):
> 
> "I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"
> 
> Thanks,
> 
> The AVM Core Team
> 
> #RR

"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

**Comment by matebarabas** [comment]
Hi @araybiajenkins,

Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!

Before starting development, please ensure ALL the following requirements are met.

**Please use the following values explicitly as provided in the [module index](https://azure.github.io/Azure-Verified-Modules/indexes/) page**:

- For your module:
  - `ModuleName` - for naming your module
- For your module's repository:
  - Repo name and folder path are defined in `RepoURL`

Check if this module exists in the other IaC language. If so, collaborate with the other owner for consistency. 👍

You can now start the development of this module! ✅ Happy coding! 🎉

**Please respond to this comment and request a review from the AVM core team once your module is ready to be published! Please include a link pointing to your PR, once available. 🙏**

Any further questions or clarifications needed, let us know!

Thanks,

The AVM Core Team
